src/repository.cpp

```diff
@@ -897,7 +924,8 @@ void FastImportRepository::Transaction::commit()
     Q_ASSERT(mark < repository->next_file_mark - 1);
 
     // create the commit message
-    QByteArray message = log;
+    // Optionally filter the log message through external program
+    QByteArray message = repository->msgFilter(log);
```

Comment:
Why not filtering the message four lines further down?
Recently I've seen the requirement that someone wanted to have the revision in the first line of the log message. With this patch this would easily be possible if the filter process is applied after the metadata was added.